catalog/core/catalog-core-solrcommands/src/test/java/org/codice/ddf/commands/solr/TestBackupCommand.java

```diff
@@ -135,13 +211,276 @@ protected HttpWrapper getHttpClient() {
+                        INVALID_COLLECTION_NAME,
+                        backupCommand.backupLocation,
+                        INVALID_COLLECTION_NAME)));
+        assertThat(consoleOutput.getOutput(),
+                containsString(String.format(
+                        "Backup failed. Unable to optimize collection [%s]. Collection not found: %s",
+                        INVALID_COLLECTION_NAME,
+                        INVALID_COLLECTION_NAME)));
+    }
+
+    @Test
+    public void testPerformSolrCloudSynchronousBackupNumberToKeepSupplied() throws Exception {
+
+        // Setup exception expectations
+        expectedException.expect(IllegalArgumentException.class);
+        expectedException.expectMessage("Invalid Argument(s). Run solr:backup --help for usage details.");
+
+        // Set system properties
+        setupSystemProperties(SOLR_CLOUD_CLIENT_TYPE);
+
+        // Setup BackupCommand
+        BackupCommand backupCommand = getSynchronousBackupCommand(getBackupLocation(), DEFAULT_CORE_NAME);
+        backupCommand.numberToKeep = 3;
+
+        // Perform Test
+        backupCommand.doExecute();
+    }
+
+    @Test
+    public void testPerformSolrCloudAsynchronousBackupWithAsyncStatusOptionsSupplied() throws Exception {
+
+        // Setup exception expectations
+        expectedException.expect(IllegalArgumentException.class);
+        expectedException.expectMessage("Invalid Argument(s). Run solr:backup --help for usage details.");
+
+        // Set system properties
+        setupSystemProperties(SOLR_CLOUD_CLIENT_TYPE);
+
+        // Setup BackupCommand
+        BackupCommand backupCommand = getBackupCommand(getBackupLocation(), DEFAULT_CORE_NAME, true, true, "myRequestId1");
+
+        // Perform Test
+        backupCommand.doExecute();
+    }
+
+    @Test
+    public void testPerformSolrCloudAsynchronousBackupNumberToKeepSupplied() throws Exception {
+
+        // Setup exception expectations
+        expectedException.expect(IllegalArgumentException.class);
+        expectedException.expectMessage("Invalid Argument(s). Run solr:backup --help for usage details.");
+
+        // Set system properties
+        setupSystemProperties(SOLR_CLOUD_CLIENT_TYPE);
+
+        // Setup BackupCommand
+        BackupCommand backupCommand = getAsnychronousBackupCommand(getBackupLocation(), DEFAULT_CORE_NAME);
+        backupCommand.numberToKeep = 3;
+
+        // Perform Test
+        backupCommand.doExecute();
+    }
+
+    @Test
+    public void testPerformSolrCloudAsynchronousBackupAndStatus() throws Exception {
+
+        // Set system properties
+        setupSystemProperties(SOLR_CLOUD_CLIENT_TYPE);
+
+        // Setup BackupCommand for async backup
+        BackupCommand backupCommand = getAsnychronousBackupCommand(getBackupLocation(), DEFAULT_CORE_NAME);
+
+        // Perform Test (backup)
+        backupCommand.doExecute();
+
+        // Verify
+        assertThat(consoleOutput.getOutput(),
```

Comment:
❗️We should split this test in two, one to test the async backup on its own and one to test the status part.